09/Time.java

```diff
@@ -0,0 +1,30 @@
+public class Time {
+    private int hours;
+    private int minutes;
+
+    public Time(int hours, int minutes) {
+        if (hours > 0 && hours < 24) {
+            this.hours = hours;
+        } else {
+            //throw
+        }
+        if (minutes > 0 && minutes < 60) {
+            this.minutes = minutes;
+        } else {
+            //throw
+        }
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (!(obj instanceof Time)) {
+            return false;
+        }
+        Time other = (Time) obj;
+        return (other.hours == this.hours) && (other.minutes == this.minutes);
+    }
+
+    public String toString() {
+        return String.format("%s : %s", hours, minutes);
```

Comment:
```suggestion
        return String.format("%d : %d", hours, minutes);
```
When formatting: `%s` is for strings, `%d` is integers.

Reply:
hoo, shoot!
I thought it was the other way around